bridges/snowbridge/pallets/system/src/lib.rs

```diff
@@ -243,6 +253,16 @@ pub mod pallet {
 	pub type PricingParameters<T: Config> =
 		StorageValue<_, PricingParametersOf<T>, ValueQuery, T::DefaultPricingParameters>;
 
+	#[pallet::storage]
+	#[pallet::getter(fn tokens)]
+	pub type Tokens<T: Config> =
+		StorageMap<_, Twox64Concat, TokenId, VersionedLocation, OptionQuery>;
```

Comment:
The stored locations should not be versioned. Instead when XCMv5 comes out, we add a storage migration.

`VersionedLocation` is more suitable for parameters passed to extrinsics, to ensure the extrinsic ABI/API compatibility does not to be broken when new XCM versions come out.

Reply:
I would assume `XCMv5` is backward compatible so with `VersionedLocation` there is no need for storage migration.

Even if it's not, I don't see any issue migrate from a `VersionedLocation(V4)`.

Reply:
In my opinion, `VersionedLocation` is not meant to be used for private storage fields. For example the `ForeignAssets` pallet on AH stores raw `Locations` for assets. 

As a general principle, its desirable for data persisted in storage to be consistent, regular, and ordered (ie. lacking in entropy). This makes it easier to process and reason about the data.

`VersionedLocation` is the opposite of that, it introduces disorder. Its only use case is for public APIs that should not break when new XCM versions are added. 

Additionally, using `VersionedLocation` for storage fields does not prevent the need for migrations. Gav said himself that older XCM versions will be removed. As an example, there is no ``VersionedLocation::V1` variant.

Reply:
https://github.com/Snowfork/polkadot-sdk/pull/128/commits/2675241027a6234971361e4b2424e86e55978091